examples/hosts/app-integration/external-data/src/mock-external-data-service/externalDataSource.ts

```diff
@@ -106,7 +106,7 @@ export class ExternalDataSource extends TypedEventEmitter<IExternalDataSourceEve
 		this.data[externalTaskListId] = data;
 
 		// Emit for debug views to update
-		this.emit("debugDataWritten", this.data);
+		this.emit("debugDataWritten", this.data, externalTaskListId);
```

Comment:
Nit: I think the id as first parameter feels a bit cleaner.

But more importantly, this is writing an `ITaskListData` that contains all task lists, when the event says the argument will be an `ITaskData` (which should be the specific task list data that corresponds to the accompanying id in the event). Our typing can't help much when emitting the event and didn't catch this unfortunately. With this fixed, you'll be able to simplify one line in webhook.ts which currently has to grab a specific task list out of the `ITaskListData` it's receiving.

(And wait for the next comment, where I suggest we need to take all the new logic out of webhook.ts 😄 ).